src/main/java/seedu/partyplanet/logic/commands/EditCommand.java

```diff
@@ -1,258 +1,37 @@
 package seedu.partyplanet.logic.commands;
 
-import static java.util.Objects.requireNonNull;
+import static seedu.partyplanet.logic.parser.CliSyntax.FLAG_REMOVE;
 import static seedu.partyplanet.logic.parser.CliSyntax.PREFIX_ADDRESS;
 import static seedu.partyplanet.logic.parser.CliSyntax.PREFIX_BIRTHDAY;
 import static seedu.partyplanet.logic.parser.CliSyntax.PREFIX_EMAIL;
 import static seedu.partyplanet.logic.parser.CliSyntax.PREFIX_NAME;
 import static seedu.partyplanet.logic.parser.CliSyntax.PREFIX_PHONE;
 import static seedu.partyplanet.logic.parser.CliSyntax.PREFIX_REMARK;
 import static seedu.partyplanet.logic.parser.CliSyntax.PREFIX_TAG;
-import static seedu.partyplanet.model.Model.PREDICATE_SHOW_ALL_PERSONS;
-
-import java.util.Collections;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Optional;
-import java.util.Set;
-
-import seedu.partyplanet.commons.core.Messages;
-import seedu.partyplanet.commons.core.index.Index;
-import seedu.partyplanet.commons.util.CollectionUtil;
-import seedu.partyplanet.logic.commands.exceptions.CommandException;
-import seedu.partyplanet.model.Model;
-import seedu.partyplanet.model.person.Address;
-import seedu.partyplanet.model.person.Birthday;
-import seedu.partyplanet.model.person.Email;
-import seedu.partyplanet.model.person.Name;
-import seedu.partyplanet.model.person.Person;
-import seedu.partyplanet.model.person.Phone;
-import seedu.partyplanet.model.person.Remark;
-import seedu.partyplanet.model.tag.Tag;
 
 /**
  * Edits the details of an existing person in PartyPlanet.
  */
-public class EditCommand extends Command {
+public abstract class EditCommand extends Command {
 
     public static final String COMMAND_WORD = "edit";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the person identified "
             + "by the index number used in the displayed person list. "
             + "Existing values will be overwritten by the input values.\n"
-            + "Parameters: INDEX (must be a positive integer) "
             + "[" + PREFIX_NAME + " NAME] "
             + "[" + PREFIX_PHONE + " PHONE] "
             + "[" + PREFIX_EMAIL + " EMAIL] "
             + "[" + PREFIX_ADDRESS + " ADDRESS] "
             + "[" + PREFIX_TAG + " TAG]... "
             + "[" + PREFIX_BIRTHDAY + " BIRTHDAY] "
             + "[" + PREFIX_REMARK + " REMARK]\n"
+            + "OR " + COMMAND_WORD + " " + FLAG_REMOVE + ": Removes specified tag from all persons.\n"
```

Comment:
Might have to consider a different phrasing later, once the `--append` flag for tags is added in as well. Should be fine for now.

Reply:
```suggestion
            + "OR " + COMMAND_WORD + " " + FLAG_REMOVE + ": Removes specified tag from all people in the filtered list.\n"
```

Reply:
Will leave it to after append is implemented to see how they fit.